rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,269 @@
+from pathlib import PurePath
+from typing import Dict, Text, List, Any, Optional, Union
+
+import rasa.utils.io
+from rasa.constants import DOCS_URL_STORIES
+from rasa.core.constants import INTENT_MESSAGE_PREFIX
+from rasa.core.events import UserUttered, SlotSet
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.structures import StoryStep
+from rasa.data import YAML_FILE_EXTENSIONS
+import rasa.utils.common as common_utils
+
+KEY_STORIES = "stories"
+KEY_STORY_NAME = "story"
+KEY_STORY_STEPS = "steps"
+KEY_STORY_USER_UTTERANCE = "user"
+KEY_STORY_ENTITIES = "entities"
+KEY_SLOT_NAME = "slot"
+KEY_SLOT_VALUE = "value"
+KEY_ACTION = "action"
+KEY_CHECKPOINT = "checkpoint"
+KEY_CHECKPOINT_SLOTS = "slots"
+KEY_METADATA = "metadata"
+KEY_OR = "or"
+
+
+class YAMLStoryReader(StoryReader):
+    """Class that reads the core training data in a YAML format
+
+    """
+
+    async def read_from_file(self, filename: Text) -> List[StoryStep]:
+        try:
+            yaml_content = rasa.utils.io.read_yaml_file(filename)
+            if not isinstance(yaml_content, dict):
+                common_utils.raise_warning(
+                    f"Failed to read {filename}. It should be a Yaml dict."
+                )
+                return []
+
+            stories = yaml_content.get(KEY_STORIES)  # pytype: disable=attribute-error
+            if not stories:
+                return []
+
+            self._parse_stories(stories)
+            self._add_current_stories_to_result()
+            return self.story_steps
+
+        except ValueError as e:
+            common_utils.raise_warning(
+                f"Failed to read {filename}, it will be skipped. Error: {e}"
+            )
+
+        return []
+
+    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
+        for story_item in stories:
+            if not isinstance(story_item, dict):
+                common_utils.raise_warning(
+                    f"Unexpected block found in '{self.source_name}': \n"
+                    f"{story_item}\n"
+                    f"Items under the `{KEY_STORIES}` key must be YAML dictionaries."
+                    f"It will be skipped.",
+                    docs=DOCS_URL_STORIES,
+                )
+                continue
+
+            if KEY_STORY_NAME in story_item.keys():
+                self._parse_plain_story(story_item)
+
+    def _parse_plain_story(self, story_item: Dict[Text, Any]) -> None:
+        story_name = story_item.get(KEY_STORY_NAME, "")
+        if not story_name:
+            common_utils.raise_warning(
+                f"Issue found in '{self.source_name}': \n"
+                f"{story_item}\n"
+                f"The story has an empty name. "
+                f"Stories should have a name defined under `{KEY_STORY_NAME}` key. "
+                "It will be skipped.",
+                docs=DOCS_URL_STORIES,
+            )
+
+        steps = story_item.get(KEY_STORY_STEPS, "")
```

Comment:
? + should there be a warning if there are not steps?
```suggestion
        steps = story_item.get(KEY_STORY_STEPS, [])
```